docs/Policies.md

```diff
@@ -100,14 +100,15 @@ in your policy:
 ```php
 namespace App\Policy;
 
-use Authorization\Policy\BeforePolicyInterface;
+use Phauthentic\Authorization\Policy\Result;
+use Phauthentic\Authorization\Policy\BeforePolicyInterface;
 
 class ArticlesPolicy implements BeforePolicyInterface
 {
     public function before($user, $resource, $action)
     {
         if ($user->getOriginalData()->is_admin) {
-            return true;
+            return new Result(true);
```

Comment:
Is there no longer support for returning bare booleans?

Reply:
Yes support for boolean return will be removed in next major. Instance of `ResultInterface` must always be returned now. This allows for cleaner API and return type hinting.

I'll fix the docs wherever it still mentions boolean return.

Reply:
Doesn't always requiring Result objects make both building Policies and using can() harder more awkward though?

Could we add methods like `Result::success()` and `Result::failure()` as porcelain methods for making pass/fail result objects?

Reply:
You mean static methods?

Reply:
Yes. Having to wrap booleans is a lot of syntax. I'd prefer to not break that interface as it doesn't benefit the end user at all. I understand it makes typehinting easier, but its at the detriment of usability.

Reply:
I agree that having to create an object instead of returning boolean when one doesn't need to set a custom reason (which would be majority of time) is not user friendly.

We could continue allowing policy classes to return boolean or result object.

But for `IdentifyInterface::can()` and `AuthorizationInterface::can()` always returning objects is better as then one doesn't have to check return value type in app code. Implementation of these method could handle creating of result object from booleans if required.

/cc @robertpustulka @burzum

Reply:
AFAIR policy classes were not required to return objects. I agree that’s not necessary. Identity and service should return objects only for consistency though.

Reply:
Instead of forcing result objects to be used in can() should we consider exposing another method that returns result objects?

My concern is that we will be breaking the core methods of this plugin and adding complexity everywhere when it is rarely necessary. Having can() return bool|Result was a mistake in hindsight. We could correct that erring on the side of simplicity instead of adding complexity like we would be doing with this change.

Reply:
I have made a PR https://github.com/Phauthentic/authorization/pull/4 to allow policy method to return boolean again.

Reply:
> My concern is that we will be breaking the core methods of this plugin and adding complexity everywhere when it is rarely necessary.

Isn't worrying about breaking methods for the plugin which had it's 1st stable release (prematurely IMO) only few months ago being overly conservative?

This discussion also makes me second guess whether using `phauthetic/authorization` as a dependency is a good idea. The goals of this plugin and the agnostic lib could diverge in future causing more friction with regards to the API.

My initial goal was to have the plugin ready for Cake 4 which has already been achieve with #76.

Reply:
Even if we were pre 1.0 I wouldn't like the change to adopt result objects everywhere. I feel it adds complexity and overhead where it isn't useful in the majority of scenarios.

Reply:
I have no interest in continuing this discussion and am fine simply closing this PR.

As I mentioned 2.x branch is already Cake 4 ready.